workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/GameObjectWorldCommandSystem.cs

```diff
@@ -0,0 +1,138 @@
+using System;
+using System.Linq;
+using Improbable.Gdk.Core.Commands;
+using Unity.Collections;
+using Unity.Entities;
+
+#region Diagnostic control
+
+// Disable the "variable is never assigned" for injected fields.
+#pragma warning disable 649
+
+// ReSharper disable ClassNeverInstantiated.Global
+
+#endregion
+
+
+namespace Improbable.Gdk.Core.GameObjectRepresentation
+{
+    public class GameObjectWorldCommandSystem : ComponentSystem
+    {
+        private struct ReserveEntityIdsResponseData
+        {
+            [ReadOnly] public readonly int Length;
+            [ReadOnly] public EntityArray Entities;
+            [ReadOnly] public ComponentDataArray<WorldCommands.ReserveEntityIds.CommandResponses> CommandResponses;
+            [ReadOnly] public ComponentArray<GameObjectReference> HasGameObjectReference;
+        }
+
+        private struct CreateEntityResponseData
+        {
+            [ReadOnly] public readonly int Length;
```

Comment:
I'm pretty sure having `[ReadOnly]` here is not allowed. that only works on collections

Reply:
It didn't complain but I am guessing it's a noop, will clean up